client/components/app-promo/index.jsx

```diff
@@ -104,9 +79,31 @@ export default React.createClass( {
 						height="32"
 						alt="WordPress Desktop Icon"
 					/>
-					{ this.state.promo_item.message }
+					{ message }
 				</a>
 			</div>
 		);
 	}
 } );
+
+AppPromo.defaultProps = {
+	promoItem: getRandomPromo(),
+	getPromoLink: ( location, promoCode ) => {
+		return getPromoLink( location, promoCode );
+	},
```

Comment:
this is probably a good place to simplify. we can just return `getPromoLink` since the parameters are identical. All we're doing by creating the closure is adding overhead…

```js
getPromoLink: getPromoLink
```

…although we realize now that we can use the object shorthand notation and leave off the value since the key and value have the same name…

```js
AppPromo.defaultProps = {
	promoItem: getRandomPromo(),
	translate: identity,
	recordTracksEvent: noop,
	saveDismissal: () => store.set( 'desktop_promo_disabled', true ),
	getPromoLink,
}
```

also I inlined `store.set()` here since we don't need the curlies around it